packages/devtools_app/lib/src/memory/memory_protocol.dart

```diff
@@ -248,6 +249,22 @@ class MemoryTracker {
     }
   }
 
+  /// Grab current EventSample, if any extensionEvents add those to this
+  /// EventSample and clear the outstanding extension events.
+  EventSample pullCloneClear(MemoryTimeline memoryTimeline, int time) {
```

Comment:
Can you give this a clearer name that indicates the task done not how it is implemented?

Reply:
```
  /// Many extension events could arrive between memory collection ticks, those
  /// events need to be associated with a particular memory tick (timestamp). This
  /// routine collects those new events received that are closest to a tick
  /// (time parameter)).
  /// @returns copy of events to associate with an existing HeapSample tick
  /// (contained in the EventSample). See [processEventSample] it computes the
  /// events to aggregate to an existing HeapSample or delay associating those
  /// events until the next HeapSample (tick) received see [_recalculate].
  EventSample pullClone(MemoryTimeline memoryTimeline, int time) {

```